avocado/core/varianter.py

```diff
@@ -459,6 +459,57 @@ def data_merge(self, tree):     # pylint: disable=E0202
             return
         self.data.merge(tree)
 
+    def str_variants(self):
```

Comment:
As I mentioned in previous PRs this code is rewritten almost from scratch. That means there are way too many changes. I did update the existing unittests, but I plan to write many more when the API settles. I asked you to be lenient towards the lack of unittest/documentation updates because I know what else is coming and we are almost there. This series actually includes the major rework of the documentation (which I kept promising in previous stages) and when we agree on this approach (I still haven't got ack on the base idea...) I can write the unit and functional tests.

Although the main functionality is already covered so I need to extend just the corner cases...